### PR TITLE
Revert : conformance: promote container exec probe timeout tests

### PR DESCRIPTION
**Pull request by dims**
Reverting af3e118b1fc4c90c088da9ffa375087b1dcf947f and
2242d0ffc409594bd4543153ebdba4534c8e83cf as these tests fail when
ExecProbeTimeout feature gate is turned on.

Signed-off-by: Davanum Srinivas <davanum@gmail.com>

#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
We promoted a conformance test that is not yet ready. The behavior of the conformance test changes with feature flags applied.

fails the first test `it tests only GA, non-optional features or APIs (e.g., no alpha or beta endpoints, no feature flags required, no deprecated features)` from the guidelines:
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md#conformance-test-requirements

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:
Incorporated changes from https://github.com/kubernetes/kubernetes/pull/100028 by @jackfrancis 

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/e2e/common/node/container_probe.go [hunk 1]
```diff
@@ -213,7 +214,10 @@ var _ = SIGDescribe("Probing container", func() {
 		Testname: Pod liveness probe, container exec timeout, restart
 		Description: A Pod is created with liveness probe with a Exec action on the Pod. If the liveness probe call does not return within the timeout specified, liveness probe MUST restart the Pod.
 	*/
-	framework.ConformanceIt("should be restarted with an exec liveness probe with timeout [NodeConformance]", func() {
+	ginkgo.It("should be restarted with an exec liveness probe with timeout [NodeConformance]", func() {
```
Comment (chewong) [inline test/e2e/common/node/container_probe.go:217]: Does it also make sense to remove NodeConformance?

Comment (dims) [reply test/e2e/common/node/container_probe.go:217]: @chewong nope. this is still useful in `NodeConformance`.

Comment (dims) [reply test/e2e/common/node/container_probe.go:217]: we are also checking the flag now. so should work fine.

**Comment by derekwaynecarr** [comment]
Thanks @dims !

/lgtm
/approve

**Comment by Riaankl** [comment]
@dims I looked that the endpoint tested by these tests and they are all touched by several other conformance tests.
This should not affect total Conformance cover.

**Comment by oomichi** [comment]
Thanks for doing this @dims 

/lgtm

**Comment by Riaankl** [comment]
APISnoop was updated after this PR merged and there is no loss in conformance cover as expected.

**Comment by liggitt** [comment]
> Reverting [af3e118](https://github.com/kubernetes/kubernetes/commit/af3e118b1fc4c90c088da9ffa375087b1dcf947f) and
> [2242d0f](https://github.com/kubernetes/kubernetes/commit/2242d0ffc409594bd4543153ebdba4534c8e83cf) as these tests fail when
> ExecProbeTimeout feature gate is turned on.

is that accurate? the gate is on by default, and conformance tests were passing, right?

**Comment by liggitt** [comment]
got it 👍

**Comment by chewong** [comment]
The test case is still running with `ExecProbeTimeout` explicitly in the kubelet config (see https://testgrid.k8s.io/provider-azure-upstream#k8s-e2e-conformance-master). Are there any extra configurations I should be setting?

**Comment by liggitt** [comment]
looks like that job is still running tests that are not tagged with `[Conformance]` ... https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-e2e-aks-engine-conformance-master/1369672142546800640

**Comment by liggitt** [comment]
`--test_args=--ginkgo.focus=\[Conformance\]|\[NodeConformance\]`

**Comment by BenTheElder** [comment]
> --test_args=--ginkgo.focus=\[Conformance\]|\[NodeConformance\]

conformance e2e (as in the tests we see as conformance with the CNCF) is typically just `--test_args=--ginkgo.focus=\[Conformance\]`, unless something changed ...

**Pull request merged**